front/commonitilvalidation.form.php

```diff
@@ -98,5 +98,21 @@
       ]);
       Html::back();
    }
+} else if (isset($_REQUEST['delete_document'])) {
+   $ticket = new Ticket();
+   $ticket->getFromDB((int)$_REQUEST['tickets_id']);
+   $doc = new Document();
+   $doc->getFromDB(intval($_REQUEST['documents_id']));
+   if ($doc->can($doc->getID(), UPDATE)) {
+      $document_item = new Document_Item;
+      $found_document_items = $document_item->find([
+         $ticket->getAssociatedDocumentsCriteria(),
+         'documents_id' => $doc->getID()
+      ]);
+      foreach ($found_document_items  as $item) {
+         $document_item->delete(Toolbox::addslashes_deep($item), true);
+      }
+   }
+   Html::back();
```

Comment:
This logic is already implemented in `front/ticket.form.php`. You could directly call method on this script instead of copy/paste it here.